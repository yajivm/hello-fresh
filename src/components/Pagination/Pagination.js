import React from 'react';
import PropTypes from 'prop-types';
import { BackIc, ForwardIc } from "../Icons";
import { PaginatioNumber, PaginationWrapper, Icon } from './Pagination.style';

const Pagination = ({
  page,
  showBackIc,
  showForwardIc,
  onClickBackIc,
  onClickForwardIc
}) => {

  return (
    <PaginationWrapper data-testid="pagination-wrapper">
      <Icon data-testid="back-ic" onClick={() => onClickBackIc('Back')} isIcToBeShown={showBackIc}>
        <BackIc />
      </Icon>
      <PaginatioNumber data-testid="page-number-title">{page}</PaginatioNumber>
      <Icon data-testid="forward-ic" onClick={() => onClickForwardIc('Forward')} isIcToBeShown={showForwardIc}>
        <ForwardIc />
      </Icon>
    </PaginationWrapper>
  )
};

Pagination.propTypes = {
  page: PropTypes.any.isRequired,
  onClickForwardIc: PropTypes.func.isRequired,
  onClickBackIc: PropTypes.func.isRequired,
  showBackIc: PropTypes.bool,
  showForwardIc: PropTypes.bool,
}

Pagination.defaultProps = {
  showBackIc: false,
  showForwardIc: false,
}

export default Pagination;