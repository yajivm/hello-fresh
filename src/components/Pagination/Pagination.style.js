import styled from "styled-components";

export const PaginationWrapper = styled.div`
  display: flex;
  text-align: center;
  align-items: center;
  justify-content: center;
  margin-bottom: 40px;
`

export const Icon = styled.div`
  width: 45px;
  height: 35px;
  background: #eee;
  border-radius: 50%;
  padding-top: 12px;
  visibility: ${({ isIcToBeShown }) => isIcToBeShown ? 'visible' : 'hidden'}
`

export const PaginatioNumber = styled.p`
  margin: 0 20px;
  font-size: 20px;
  font-weight: 900;
`;