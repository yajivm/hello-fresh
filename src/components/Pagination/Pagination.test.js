import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import Pagination from '.'

const mockClickBackIc = jest.fn();
const mockClickForwardIc = jest.fn();
const props = {
  page: 'title',
  showBackIc: true,
  showForwardIc: true,
  onClickBackIc: mockClickBackIc,
  onClickForwardIc: mockClickForwardIc,
}

describe('Pagination', () => {

  it('should render the component', () => {
    const { container } = render(<Pagination {...props} />);

    expect(screen.getByTestId('page-number-title')).toHaveTextContent('title');
    expect(container).toMatchSnapshot();
  })

  it('should not show icons when prop values are false', () => {
    const customProps = {
      page: 'title',
      onClickBackIc: mockClickBackIc,
      onClickForwardIc: mockClickForwardIc,
    };
    render(<Pagination {...customProps} />);

    expect(screen.queryByTestId('back-ic')).toHaveStyle({ visibility: 'hidden' });
    expect(screen.queryByTestId('forward-ic')).toHaveStyle({ visibility: 'hidden' });
  })

  it('should trigger callback methods when icons been clicked', () => {
    render(<Pagination {...props} />);

    fireEvent.click(screen.getByTestId('back-ic'));

    expect(mockClickBackIc).toHaveBeenCalledTimes(1);

    fireEvent.click(screen.getByTestId('forward-ic'));

    expect(mockClickForwardIc).toHaveBeenCalledTimes(1);
  })
});