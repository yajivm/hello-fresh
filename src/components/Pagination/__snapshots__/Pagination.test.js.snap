// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render the component 1`] = `
<div>
  <div
    class="sc-bczRLJ kOjpRg"
    data-testid="pagination-wrapper"
  >
    <div
      class="sc-gsnTZi hiWrvB"
      data-testid="back-ic"
    >
      <svg
        fill="currentColor"
        height="1.5em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="1.5em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M9.707 16.707a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414l6-6a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l4.293 4.293a1 1 0 010 1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <p
      class="sc-dkzDqf bwSjOE"
      data-testid="page-number-title"
    >
      title
    </p>
    <div
      class="sc-gsnTZi hiWrvB"
      data-testid="forward-ic"
    >
      <svg
        fill="currentColor"
        height="1.5em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="1.5em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;
